protocols/ads/src/main/resources/protocols/ads/ads.mspec

```diff
@@ -386,7 +386,7 @@
     [array int 8 'data' count 'sampleSize']
 ]
 
-[dataIo 'DataItem' [string 'dataFormatName', int 32 'stringLength']
+[dataIo 'DataItem' [string '-1' 'dataFormatName', int 32 'stringLength']
```

Comment:
what does '-1' mean for a string?

Reply:
'-1' means that it is a null terminated string, which was the default previously.
We can't have an optional field here using expression as we already have the encoding field straight after it.